File: sol/2483.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        n = len(customers)
        bestTime = n
        bestPen = n
        p = n
        for i in range(n - 1, -1, -1):
            if customers[i] == 'Y':
                p += 1
            else:
                p -= 1
                if p <= bestPen:
                    bestPen = p
                    bestTime = i
        return bestTime

File: sol/2483/2483.cpp
struct Solution {
    int bestClosingTime(string &customers) {
        int n = customers.size();
        int bestTime = 0, bestPen = 0, p = 0;
        for (int i = 0; i < n; ++i){
            p += 1 - 2 * (customers[i] == 'Y');
            if (p < bestPen){
                bestPen = p;
                bestTime = i + 1;
            }
        }
        return bestTime;
    }
};
